### PR TITLE
Document layout in Storybook

### PR DESCRIPTION
**Pull request by jameskoster**
Screenshot 2024-12-05 at 12 42 50

## What?
Replicates the documentation on Layout we find [in Figma](https://www.figma.com/design/804HN2REV2iap2ytjRQ055/WordPress-Design-System?node-id=15991-23086&t=zp5VJC3UkCO3wnEh-1) on Storybook.

## Why?
Continues the expansion of foundation documentation.

## Testing Instructions
1. `npm run storybook:dev`
2. Navigate to the [Foundations / Layout page](http://localhost:50240/?path=/docs/foundations-layout--page).

Now would be a good moment to review this documentation.

In future we can expand this to include more detail. For instance we might describe compositions for section headers, layout grids, and more.

### CODE REVIEW

**Review by mirka — COMMENTED** [review]
Not a blocker, but how much do you anticipate the static images needing to be updated throughout the life of this page?

If we do anticipate updates, it may be more maintainable to render the visuals with code (i.e. divs with CSS styles) or by Figma embed. Otherwise the person wanting to update the images would need to track you down for the source files 😄

**Comment by jasmussen** [comment]
> Not a blocker, but how much do you anticipate the static images needing to be updated throughout the life of this page?

Related to that, I recall early in the life of this project that we added a lot of images for documentation, which caused trouble for contributors having to pull down big PNG files in multiple versions. I wonder how we can avoid that happening again.

File: storybook/stories/foundations/layout.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Layout defines the arrangement and structure of admin pages, content sections, and other visual elements.
+
+## Page areas
+
+At the highest level admin pages are comprised of _areas_, that can be arranged in different ways.
+
+<img src={ areas } alt="Area definition diagram" width="100%" />
+
+1. **Sidebar** – Primary admin navigation UI including notifications and access to Command Palette. Always positioned on the left of the page (assuming LTR locale).
```
Comment (juanbuis) [inline storybook/stories/foundations/layout.mdx]: Perhaps use "left-to-right language" or "left-to-right locale" to avoid the acronym?

**Review by juanbuis — COMMENTED** [review]
Looks great! Left one comment :)

**Comment by jameskoster** [comment]
> Otherwise the person wanting to update the images would need to track you down for the source files 😄

They're SVGs, so they effectively _are_ the source files. Unless I'm missing a beat?

Edit: They also exist in the [WPDS source file](https://www.figma.com/design/804HN2REV2iap2ytjRQ055/WordPress-Design-System?node-id=16339-39747&t=zp5VJC3UkCO3wnEh-1), so all members of the design team should have access, or be able to easily get access.

**Comment by mirka** [comment]
> They're SVGs, so they effectively _are_ the source files. Unless I'm missing a beat?
> 
> Edit: They also exist in the [WPDS source file](https://www.figma.com/design/804HN2REV2iap2ytjRQ055/WordPress-Design-System?node-id=16339-39747&t=zp5VJC3UkCO3wnEh-1), so all members of the design team should have access, or be able to easily get access.

Fair! Let's go with this and we can always revisit if we encounter maintenance issues.

**Review by mirka — APPROVED** [review]

**Comment by jameskoster** [comment]
@mirka do you know if it's okay to merge this? I'm seeing some failing checks but they seem optional/unrelated.

**Comment by mirka** [comment]
@jameskoster Looks good to me 👍

**Pull request merged**